iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/gcp/PublicAccessCheck.java

```diff
@@ -71,8 +80,25 @@ protected void registerResourceConsumer() {
       resource -> resource.attribute("visibility")
         .reportIf(equalTo("public"), MESSAGE)
         .reportIfAbsent(OMITTING_DNS));
-  }
 
+    register("google_container_cluster",
+      resource -> {
+        BlockSymbol config = resource.block("private_cluster_config");
+        config.reportIfAbsent(OMITTING_KUBERNETES);
+        AttributeSymbol nodes = config.attribute("enable_private_nodes");
+        AttributeSymbol endpoint = config.attribute("enable_private_endpoint");
+
+        if (!(nodes.isPresent() || endpoint.isPresent())) {
+          config.report(String.format(OMITTING_KUBERNETES, "enable_private_nodes and enable_private_endpoint"));
+        } else if (nodes.is(isFalse()) && endpoint.is(isFalse())) {
+          nodes.report(MESSAGE_KUBERNETES, endpoint.toSecondary(MESSAGE_KUBERNETES));
+        } else {
+          Stream.of(nodes, endpoint)
+            .forEach(symbol -> symbol.reportIf(isFalse(), MESSAGE_KUBERNETES)
+              .reportIfAbsent(OMITTING_KUBERNETES));
+        }
```

Comment:
This is such a "state of the art" **if - else if - else**  !
I've changed my mind over it correctness at least twice ...
Kudos 😅

Reply:
It does the job;
I am only suspecting that its cognitive complexity is nontrivial.
But no - it is not necessary to address this.